Druid/DRDominatorTree.class.st

```diff
@@ -148,42 +188,81 @@ DRDominatorTree >> dominatorOfAll: instructions [
 				  and: (self blockDictionary at: instruction basicBlock) ]
 ]
 
-{ #category : #asd }
+{ #category : #initialization }
+DRDominatorTree >> findImmediateDominatorOfBlock: nextBlocksToIterate [
+	"Method used during the tree construction, don't use it outside of it"
+
+	| nearestCommonAncestor currentBlock |
+	nearestCommonAncestor := self blockDictionary at:
+		                         nextBlocksToIterate first.
+	nextBlocksToIterate do: [ :p |
+		currentBlock := self blockDictionary at: p.
+		nearestCommonAncestor := self
+			                         nearestCommonAncestorOf:
+			                         nearestCommonAncestor
+			                         and: currentBlock ].
+	^ nearestCommonAncestor
+]
+
+{ #category : #querying }
 DRDominatorTree >> immediateDominatorOf: aBlock [
 
 	| node |
 	node := blockDictionary at: aBlock.
 	^ node parent block
 ]
 
-{ #category : #'as yet unclassified' }
-DRDominatorTree >> initialize: cfg [
+{ #category : #initialization }
+DRDominatorTree >> initialize: cfg startingFrom: aBlockClosure [
+
+	| currentNode workList node initialBasicBlock |
+	blockDictionary := Dictionary new.
+	root := DRDominatorTreeNode new.
+	workList := OrderedCollection new.
+	workList add: root.
+
+	initialBasicBlock := aBlockClosure value: cfg.
+	root block: initialBasicBlock.
+	blockDictionary at: initialBasicBlock put: root.
+
+	[ workList isNotEmpty ] whileTrue: [
+		currentNode := workList removeFirst.
+		currentNode block successors do: [ :b |
+			blockDictionary at: b ifAbsent: [
+				node := DRDominatorTreeNode new
+					        block: b;
+					        parent: currentNode;
+					        yourself.
+				currentNode addChild: node.
+
+				blockDictionary at: b put: node.
+				workList addFirst: node ] ] ]
+]
```

Comment:
This method is used? Could be defined based on 
`DRDominatorTree >> initialize: cfg startingFrom: initialBasicBlock iterating: aBlockClosure`

Reply:
Probably not used anymore. I'll add an issue to clean it up